### PR TITLE
Upgrading Spring Data Couchbase to 4.x

### PR DESCRIPTION
**Pull request by deniswsrosa**
- Upgraded Spring Data Couchbase to 4.x
- Fixed tests accordingly

### CODE REVIEW

File: couchbase/example/src/main/java/example/springdata/couchbase/repository/AirlineRepository.java [hunk 1]
```diff
@@ -18,35 +18,31 @@
...
  *
  * @author Chandana Kithalagama
  * @author Mark Paluch
+ * @author Denis Rosa
  */
-@N1qlPrimaryIndexed
-@ViewIndexed(designDoc = "airlines")
-public interface AirlineRepository extends CrudRepository<Airline, String> {
+@Repository
+public interface AirlineRepository extends CouchbaseRepository<Airline, String> {
```
Comment (daschl) [inline couchbase/example/src/main/java/example/springdata/couchbase/repository/AirlineRepository.java:32]: I think you should still extend the PagingAndSorting (or the crud) repository, not the couchbase one directly

File: couchbase/example/src/test/java/example/springdata/couchbase/repository/AirlineRepositoryIntegrationTests.java [hunk 2]
```diff
@@ -49,20 +48,24 @@
 
 	@Autowired CouchbaseOperations couchbaseOperations;
 
+
+
 	@Before
 	public void before() {
-		airlineRepository.findById("LH").ifPresent(couchbaseOperations::remove);
+		if( couchbaseOperations.existsById().one("LH")) {
+			couchbaseOperations.removeById().one("LH");
```
Comment (daschl) [inline couchbase/example/src/test/java/example/springdata/couchbase/repository/AirlineRepositoryIntegrationTests.java:56]: is the if needed? can just always remove?

File: couchbase/pom.xml [hunk 3]
```diff
@@ -21,10 +21,32 @@
...
 		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-data-couchbase</artifactId>
+			<groupId>org.springframework.data</groupId>
+			<artifactId>spring-data-couchbase</artifactId>
+			<version>4.0.0.RC2</version>
+		</dependency>
+
+		<dependency>
+			<groupId>com.couchbase.client</groupId>
```
Comment (daschl) [inline couchbase/pom.xml:47]: this shouldn't be needed as 4.0.0 pulls it in transitively

File: couchbase/example/src/main/java/example/springdata/couchbase/repository/AirlineRepository.java (hunk 1 above)
Comment (deniswsrosa) [reply couchbase/example/src/main/java/example/springdata/couchbase/repository/AirlineRepository.java]: Fixed

File: couchbase/pom.xml (hunk 3 above)
Comment (deniswsrosa) [reply couchbase/pom.xml]: done

**Comment by mp911de** [comment]
Thanks a lot. That's merged and polished now. Once Spring Boot 2.3 is GA, the examples will go on `master`.

**Pull request closed without merging**